### PR TITLE
Move from Travis CI to Azure Pipelines

### PR DESCRIPTION
**Pull request by jb3**
yeah, that.

### CODE REVIEW

File: azure-pipelines.yml
Comment (thebeanogamer) [inline azure-pipelines.yml]: Codacy is dead, long live Codacy

Comment (jos-b) [reply azure-pipelines.yml]: is this a review comment or a reflection on codacy

Comment (thebeanogamer) [reply azure-pipelines.yml]: tenor

**Review by thebeanogamer — APPROVED** [review]

**Review by jerbob — DISMISSED** [review]
hi can you make this not azure thanks

**Comment by jos-b** [comment]
> hi can you make this not azure thanks

hi I just said "why not use a new CI solution" but beano is an easily malleable guy and he quickly set his little heart on azure and who am I to crush his dreams.

**Comment by AnonGuy** [comment]
> > hi can you make this not azure thanks
> 
> hi I just said "why not use a new CI solution" but beano is an easily malleable guy and he quickly set his little heart on azure and who am I to crush his dreams.

nay, who is he to crush the rebellion of the proletariat?

**Comment by thebeanogamer** [comment]
> 
> 
> > hi can you make this not azure thanks
> 
> hi I just said "why not use a new CI solution" but beano is an easily malleable guy and he quickly set his little heart on azure and who am I to crush his dreams.

I am a Microshill, and this seemed like the best option as a result.

**Comment by swedg** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
> 
> 
> This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

You and I are going to need to have words about your permissions level.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by swedg** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by swedg** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
WHY IS THE BUILD NOW FAILING?

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by jos-b** [comment]
hi guys always appreciate helpful discussion on the issue 😃 really glad to see we are getting somewhere with this 😄

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by jos-b** [comment]
Please can we calm this hostility 🤩

**Comment by ZomBMage** [comment]
What hostility?

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by swedg** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
Can we stop this for long enough for me to actually see why the build is failing?

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by jos-b** [comment]
> What hostility?

I just want my community dev role

**Comment by ZomBMage** [comment]
Bit shallow of a reason ngl

**Comment by jos-b** [comment]
egglien

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
I think we would all be interested in hearing @joker314's opinion here

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by AnonGuy** [comment]
This decision was made without sufficient input from community developers, and so I will leave hostile remarks until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
I think we can see that @AnonGuy has some strong opinions on this

**Review by ThisIs0xBC — APPROVED** [review]
Joseph told me to 🤷‍♂️

**Comment by thebeanogamer** [comment]
@AnonGuy because I'd prefer to get this resolved before merging, mostly so I don't have to deal with that becoming a copypasta, why are you so obsessed with circle?

**Comment by thebeanogamer** [comment]
And what specific issues do you have with Azure Devops other than Ewwww Microsoft?

**Comment by thebeanogamer** [comment]
I would suggest @swedg also does the same.

**Comment by LappySheep** [comment]
ayyyyyyyyyyyy wagwan beano my guy the legend the myth himself

**Comment by jos-b** [comment]
What is actually going on with this then

**Comment by thebeanogamer** [comment]
Discussions are occuring. I am attempting to address @swedg's concerns, but some of them do hold water, and I expect to have to contact support to resolve them.

From my perspective, Azure still seems like the best option, but my opinion is kinda irrelevant if no one is willing to contribute because I've just forced a change through.

**Comment by jos-b** [comment]
I’m willing to rewrite the CI to anything if I get community dev :smile:

**Comment by thebeanogamer** [comment]
_Takes a moment to remind you who decides who gets community dev, as I dangle the carrot in front of your nose_

**Comment by fractalpenguin** [comment]
You could write a custom CI solution in assembly, for maximum performance.

**Comment by jos-b** [comment]
> You could write a custom CI solution in assembly, for maximum performance.

beano can we do this 😃

File: azure-pipelines.yml
Comment (AnonGuy) [inline azure-pipelines.yml]: We need a `deploy` step to run `deploy.sh`.

**Review by jerbob — DISMISSED** [review]

File: azure-pipelines.yml
Comment (jos-b) [reply azure-pipelines.yml]: @AnonGuy Same deploy.sh as before?

Comment (thebeanogamer) [reply azure-pipelines.yml]: Deploy may be changing, please hold for updates, but it won't be a bash script, I can tell you that.

Comment (jos-b) [reply azure-pipelines.yml]: smh I just reverted the commit for it

**Comment by AnonGuy** [comment]
@jos-b Yeah we're considering using Ouroboros instead of updating docker containers with an ansible playbook. That way we wouldn't need to use the deploy script

**Comment by jos-b** [comment]
@thebeanogamer @AnonGuy highly advise against ourboros, it's basically API spam

**Comment by thebeanogamer** [comment]
And today on: this is why we squash and merge!

**Comment by jos-b** [comment]
Hello what is actually happening with this PR 😃 I have no clue what you boffins are coming up with 😄

**Comment by jos-b** [comment]
It'd be nice to share your plans with the PR author 😺

**Comment by AnonGuy** [comment]
wow thanks joseph!

**Comment by jos-b** [comment]
daily reminder that this exists

**Comment by thebeanogamer** [comment]
I believe this is more or less ready. Just waiting for a reply to a support ticket about moving the release pipeline to YAML and preventing builds auto-cancelling. I'll update this PR once I get a reply.

**Comment by thebeanogamer** [comment]
If it's ever finished, this will resolve #167

**Comment by jos-b** [comment]
Hey, I think we should move back again, I've been thinking about it for the past year and I don't think Azure is suiting our needs anymore.

**Comment by AnonGuy** [comment]
I'm inclined to agree with @jos-b.

**Comment by jos-b** [comment]
I'm inclined to agree with @AnonGuy.

**Comment by david-cooke** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by thebeanogamer** [comment]
You're a bad person

**Comment by david-cooke** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Comment by jchristgit** [comment]
This decision was made without sufficient input from community developers, and so I will close this until a proper discussion has been had on the topic.

**Pull request merged**
